pkg/crc/machine/start.go

```diff
@@ -606,7 +606,9 @@ func updateSSHKeyPair(sshRunner *crcssh.Runner) error {
 	}
 
 	logging.Info("Updating authorized keys...")
-	cmd := fmt.Sprintf("echo '%s' > /home/core/.ssh/authorized_keys; chmod 644 /home/core/.ssh/authorized_keys", publicKey)
+	// We are using sudo here because of https://bugzilla.redhat.com/show_bug.cgi?id=1956739
+	// MCO change this file permission to root so better to update it with sudo.
+	cmd := fmt.Sprintf("echo '%s' | sudo tee /home/core/.ssh/authorized_keys; sudo chmod 644 /home/core/.ssh/authorized_keys", publicKey)
```

Comment:
I think this can become `sshRunner.CopyData(publicKey, "/home/core/.ssh/authorized_keys", 0644)`